addons/mail/models/mail_message.py

```diff
@@ -1176,12 +1179,14 @@ def _cleanup_side_records(self):
 
     def _filter_empty(self):
         """ Return subset of "void" messages """
+        IRModel = self.env['ir.model']
         return self.filtered(
             lambda msg:
                 (not msg.body or tools.is_html_empty(msg.body)) and
                 (not msg.subtype_id or not msg.subtype_id.description) and
                 not msg.attachment_ids and
-                not msg.tracking_value_ids
+                (not IRModel._get(msg.model).is_mail_thread or
```

Comment:
Not sure we should change the heuristic here ... somehow it has tracking values so it is not "empty", is it ? It is used mainly to know if we have to "reset" message, which is rather a technical test.

Reply:
:thinking: yeah it doesn't make sense in a technical context, reverted